installation-bundle/src/Database/Installer.php

```diff
@@ -54,22 +59,22 @@ public function getCommands(): array
 
     /**
      * @throws \InvalidArgumentException
+     * @throws \Doctrine\DBAL\DBALException
      */
-    public function execCommand(string $hash): void
+    public function execCommands(array $hashes): void
     {
         if (null === $this->commands) {
             $this->compileCommands();
         }
 
-        foreach ($this->commands as $commands) {
-            if (isset($commands[$hash])) {
-                $this->connection->query($commands[$hash]);
+        $commands = [];
+        array_map(function ($c) use (&$commands): void {
+            $commands += $c;
+        }, $this->commands);
 
-                return;
-            }
+        foreach (array_intersect($this->commandOrder, $hashes) as $hash) {
+            $this->connection->query($commands[$hash]);
         }
-
-        throw new \InvalidArgumentException(sprintf('Invalid hash: %s', $hash));
```

Comment:
@leofeyer Do we need this exception? It's not catched and therefore will only ever throw on the first invalid hash. 

I left it out in the current version of the PR but can reintroduce it to either throw with a list of all invalid hashes or similarly to like it is now.

Reply:
As discussed in Mumble on November 22nd, we should check if there are invalid hashes before we even execute the first query and throw and exception.